diff --git a/Horace/herbert_core/classes/MPIFramework/@ClusterWrapper/ClusterWrapper.m b/Horace/herbert_core/classes/MPIFramework/@ClusterWrapper/ClusterWrapper.m
index 65392f0..7189a90 100644
--- a/Horace/herbert_core/classes/MPIFramework/@ClusterWrapper/ClusterWrapper.m
+++ b/Horace/herbert_core/classes/MPIFramework/@ClusterWrapper/ClusterWrapper.m
@@ -868,7 +868,20 @@ classdef ClusterWrapper
             external_dll_dir = fullfile(pths.horace, 'DLL', 'external');

             if ispc()
-                mpi_exec = fullfile(external_dll_dir, 'mpiexec.exe');
+                [rs, rv] = system('where mpiexec');
+                if rs == 0
+                    mpi_exec = strip(rv);
+                    mpis = splitlines(mpi_exec);
+                    if numel(mpis) > 1
+                        % If multiple mpiexec on path, prefer user installed MS MPI
+                        mpi_id = find(cellfun(@(x) ~isempty(strfind(x,'Microsoft')), mpis), 1);
+                        if isempty(mpi_id), mpi_id = 1; end
+                        mpi_exec = mpis{mpi_id};
+                    end
+                else
+                    % No mpiexec on path, use pre-packaged version
+                    mpi_exec = fullfile(external_dll_dir, 'mpiexec.exe');
+                end
             else
                 mpi_exec = fullfile(external_dll_dir, 'mpiexec');
             end
diff --git a/Horace/horace_core/configuration/@hpc_config/hpc_config.m b/Horace/horace_core/configuration/@hpc_config/hpc_config.m
index b3dfccf..a0e5e00 100644
--- a/Horace/horace_core/configuration/@hpc_config/hpc_config.m
+++ b/Horace/horace_core/configuration/@hpc_config/hpc_config.m
@@ -397,7 +397,7 @@ classdef hpc_config < config_base
                 pc = parallel_config();
                 wkr = pc.worker;
                 wrker_path = fileparts(which(wkr));
-                if isempty(wrker_path)
+                if isempty(wrker_path) && ~pc.is_compiled
                     warning(['HORACE:hpc_config:invalid_argument',...
                         'Can not start accumulating in separate process.',...
                         'Can not find worker on a data search path; ',...
diff --git a/Horace/horace_core/configuration/@hpc_config/private/check_worker_configured.m b/Horace/horace_core/configuration/@hpc_config/private/check_worker_configured.m
index e17e820..15c9870 100644
--- a/Horace/horace_core/configuration/@hpc_config/private/check_worker_configured.m
+++ b/Horace/horace_core/configuration/@hpc_config/private/check_worker_configured.m
@@ -5,7 +5,7 @@ function [ok,mess] = check_worker_configured(obj)
 pc = parallel_config();
 wkr = pc.worker;
 wrker_path = fileparts(which(wkr));
-if isempty(wrker_path)
+if isempty(wrker_path) && ~pc.is_compiled
     ok=false;
     mess=['Can not find worker on a data search path; ',...
         'See: http://horace.isis.rl.ac.uk/Download_and_setup#Enabling_multi-sessions_processing ',...
