diff --git a/Horace/sqw/@sqw/split.m b/Horace/sqw/@sqw/split.m
index 688c12f..11ebb1e 100644
--- a/Horace/sqw/@sqw/split.m
+++ b/Horace/sqw/@sqw/split.m
@@ -45,7 +45,11 @@ else
     nend=[];
 end
 run_contributes=false(nfiles,1);
-run_contributes(irun(nbeg))=true;   % true for runs that contribute to the data
+if min(irun) > nfiles
+    run_contributes(arrayfun(@(v) w.runid_map(v), irun(nbeg)))=true;   % true for runs that contribute to the data
+else
+    run_contributes(irun(nbeg))=true;
+end
 ind=zeros(nfiles,1);
 ind(run_contributes)=1:numel(nbeg); % index of contributing runs into nbeg and nend
 
